src/odata-to-abstract-sql.ts

```diff
@@ -992,6 +1001,35 @@ export class OData2AbstractSQL {
 			return this.ReferencedField(prop.resource, prop.name);
 		}
 	}
+
+	Method(
+		prop: unknown & {
+			method: Array<string | { method: string }>;
+		},
+	): BooleanTypeNodes | { resource: Resource; name: string } {
+		if (!prop.method) {
+			throw new SyntaxError('Method is missing method entry');
+		}
+
+		if (prop.method[0] != 'call') {
+			throw new SyntaxError(
+				`Invalid value for method invocation: ${prop.method[0]}`,
+			);
+		}
+
+		if (typeof prop.method[1] !== 'object') {
+			throw new SyntaxError(
+				`Invalid value for method invocation: ${prop.method[1]} should be an object`,
+			);
+		}
+
+		if (!this.methods.hasOwnProperty(prop.method[1].method!)) {
```

Comment:
Could you add
```suggestion
		const { method } = prop.method[1]
		if (!this.methods.hasOwnProperty(method!)) {
```
as it gets pretty long atm